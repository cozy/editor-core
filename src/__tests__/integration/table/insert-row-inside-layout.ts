import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  editable,
  animationFrame,
  getDocFromElement,
  fullpage,
} from '../_helpers';
import { insertRow } from '../../__helpers/page-objects/_table';
import { table as tableInsideLayout } from './__fixtures__/table-inside-layout';
import {
  goToEditorTestingExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';

import { clickFirstCell } from '../../__helpers/page-objects/_table';

// There was a ProseMirror bug in Safari that prevents clicks inside a selected block node,
// which caused the following integration test to fail. This was blocking the layout
// selection feature, so we skipped the test for Safari. Once that bug is fixed, we can
// re-enable the test for Safari.
// - ED-9219 https://product-fabric.atlassian.net/browse/ED-9219
// - GitHub issue https://github.com/ProseMirror/prosemirror/issues/1052
BrowserTestCase(
  'Should scale remaining columns when adding a new column preventing from going to overflow',
  { skip: ['edge', 'safari'] },
  async (client: any, testName: string) => {
    const page = await goToEditorTestingExample(client);

    await mountEditor(page, {
      appearance: fullpage.appearance,
      defaultValue: JSON.stringify(tableInsideLayout),
      allowTables: {
        advanced: true,
      },
      allowLayouts: {
        allowBreakout: true,
      },
      allowBreakout: true,
    });

    await clickFirstCell(page);
    await animationFrame(page);
    await animationFrame(page);
    await insertRow(page, 1);
    await page.type(editable, 'should be inside of the table');

    const doc = await page.$eval(editable, getDocFromElement);
    expect(doc).toMatchCustomDocSnapshot(testName);
  },
);
