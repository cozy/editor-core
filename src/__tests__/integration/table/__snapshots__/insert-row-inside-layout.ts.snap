// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should scale remaining columns when adding a new column preventing from going to overflow 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "width": 50,
          },
          "content": Array [
            Object {
              "type": "paragraph",
            },
          ],
          "type": "layoutColumn",
        },
        Object {
          "attrs": Object {
            "width": 50,
          },
          "content": Array [
            Object {
              "attrs": Object {
                "__autoSize": false,
                "isNumberColumnEnabled": false,
                "layout": "default",
              },
              "content": Array [
                Object {
                  "content": Array [
                    Object {
                      "attrs": Object {
                        "background": null,
                        "colspan": 1,
                        "colwidth": null,
                        "rowspan": 1,
                      },
                      "content": Array [
                        Object {
                          "type": "paragraph",
                        },
                      ],
                      "type": "tableHeader",
                    },
                    Object {
                      "attrs": Object {
                        "background": null,
                        "colspan": 1,
                        "colwidth": null,
                        "rowspan": 1,
                      },
                      "content": Array [
                        Object {
                          "type": "paragraph",
                        },
                      ],
                      "type": "tableHeader",
                    },
                    Object {
                      "attrs": Object {
                        "background": null,
                        "colspan": 1,
                        "colwidth": null,
                        "rowspan": 1,
                      },
                      "content": Array [
                        Object {
                          "type": "paragraph",
                        },
                      ],
                      "type": "tableHeader",
                    },
                  ],
                  "type": "tableRow",
                },
                Object {
                  "content": Array [
                    Object {
                      "attrs": Object {
                        "background": null,
                        "colspan": 1,
                        "colwidth": null,
                        "rowspan": 1,
                      },
                      "content": Array [
                        Object {
                          "content": Array [
                            Object {
                              "text": "should be inside of the table",
                              "type": "text",
                            },
                          ],
                          "type": "paragraph",
                        },
                      ],
                      "type": "tableCell",
                    },
                    Object {
                      "attrs": Object {
                        "background": null,
                        "colspan": 1,
                        "colwidth": null,
                        "rowspan": 1,
                      },
                      "content": Array [
                        Object {
                          "type": "paragraph",
                        },
                      ],
                      "type": "tableCell",
                    },
                    Object {
                      "attrs": Object {
                        "background": null,
                        "colspan": 1,
                        "colwidth": null,
                        "rowspan": 1,
                      },
                      "content": Array [
                        Object {
                          "type": "paragraph",
                        },
                      ],
                      "type": "tableCell",
                    },
                  ],
                  "type": "tableRow",
                },
                Object {
                  "content": Array [
                    Object {
                      "attrs": Object {
                        "background": null,
                        "colspan": 1,
                        "colwidth": null,
                        "rowspan": 1,
                      },
                      "content": Array [
                        Object {
                          "type": "paragraph",
                        },
                      ],
                      "type": "tableCell",
                    },
                    Object {
                      "attrs": Object {
                        "background": null,
                        "colspan": 1,
                        "colwidth": null,
                        "rowspan": 1,
                      },
                      "content": Array [
                        Object {
                          "type": "paragraph",
                        },
                      ],
                      "type": "tableCell",
                    },
                    Object {
                      "attrs": Object {
                        "background": null,
                        "colspan": 1,
                        "colwidth": null,
                        "rowspan": 1,
                      },
                      "content": Array [
                        Object {
                          "type": "paragraph",
                        },
                      ],
                      "type": "tableCell",
                    },
                  ],
                  "type": "tableRow",
                },
                Object {
                  "content": Array [
                    Object {
                      "attrs": Object {
                        "background": null,
                        "colspan": 1,
                        "colwidth": null,
                        "rowspan": 1,
                      },
                      "content": Array [
                        Object {
                          "type": "paragraph",
                        },
                      ],
                      "type": "tableCell",
                    },
                    Object {
                      "attrs": Object {
                        "background": null,
                        "colspan": 1,
                        "colwidth": null,
                        "rowspan": 1,
                      },
                      "content": Array [
                        Object {
                          "type": "paragraph",
                        },
                      ],
                      "type": "tableCell",
                    },
                    Object {
                      "attrs": Object {
                        "background": null,
                        "colspan": 1,
                        "colwidth": null,
                        "rowspan": 1,
                      },
                      "content": Array [
                        Object {
                          "type": "paragraph",
                        },
                      ],
                      "type": "tableCell",
                    },
                  ],
                  "type": "tableRow",
                },
              ],
              "type": "table",
            },
          ],
          "type": "layoutColumn",
        },
      ],
      "type": "layoutSection",
    },
  ],
  "type": "doc",
}
`;
